core/packages/analytics/pwa/components/index.js

```diff
@@ -28,39 +34,57 @@ window.dataLayer.push({
 });
 window.dataLayer.push({
   event: 'virtualPageview',
-  virtualPageview: ${JSON.stringify(gtmPageViewProperties)},
+  virtualPageview: ${JSON.stringify({ title, ...gtmPageView })},
 });`,
-      }}
-    />
-    {/* <GoogleTagManager key="GTM-K3S2BMT" id="GTM-K3S2BMT" /> */}
-    {gtmIds.map(id => <GoogleTagManager key={id} id={id} />)}
-    {gaIds.map(id => <GoogleAnalytics key={id} id={id} />)}
-    {comScoreIds.map(id => <ComScore key={id} id={id} />)}
-  </Fragment>
-);
+        }}
+      />
+      {/* <GoogleTagManager key="GTM-K3S2BMT" id="GTM-K3S2BMT" /> */}
+      {gtmIds.map(id => <GoogleTagManager key={id} id={id} />)}
+      {gaIds.map(id => (
+        <GoogleAnalytics
+          key={id}
+          id={id}
+          customDimensions={gaCustomDimensions}
+        />
+      ))}
+      {comScoreIds.map(id => <ComScore key={id} id={id} />)}
+    </Fragment>
+  );
+};
 
 Analytics.propTypes = {
   gtmIds: PropTypes.arrayOf(PropTypes.string),
   gtmClientProperties: PropTypes.shape({}),
   gtmPageViewProperties: PropTypes.shape({}),
   gaIds: PropTypes.arrayOf(PropTypes.string),
+  gaCustomDimensions: PropTypes.shape({}),
   comScoreIds: PropTypes.arrayOf(
     PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
   ),
+  title: PropTypes.string.isRequired,
 };
 
 Analytics.defaultProps = {
   gtmIds: [],
   gtmClientProperties: {},
   gtmPageViewProperties: {},
   gaIds: [],
+  gaCustomDimensions: {},
   comScoreIds: [],
 };
 
-export default inject(({ stores: { analytics } }) => ({
+const injectNotObserver = fn =>
+  compose(
+    getContext({ mobxStores: PropTypes.shape({}) }),
+    mapProps(({ mobxStores }) => fn(mobxStores)),
+  );
+
+export default injectNotObserver(({ stores: { analytics, connection } }) => ({
   gtmIds: analytics.googleTagManager.ids,
   gtmClientProperties: analytics.googleTagManager.clientProperties,
   gtmPageViewProperties: analytics.googleTagManager.pageViewProperties,
   gaIds: analytics.googleAnalytics.ids,
+  gaCustomDimensions: analytics.customDimensions(connection.selectedItem),
   comScoreIds: analytics.comScore.ids,
+  title: decode(connection.head.title).replace(/<\/?[^>]+(>|$)/g, ''),
```

Comment:
This title should come from the html <title> tag, so I think it shouldn't contain any html tag. So maybe this decode + replace is not needed. What do you think?

Reply:
@DAreRodz take a look at this (I'm not sure why it says it is outdated, because it's not).

Reply:
@luisherranz I see. However, I did that change because `connection.head.title` was returning an encoded HTML string, like `Discovering Iceland &#8211; Demo Frontity` instead of `Discovering Iceland – Demo Frontity`. Is it maybe a bug?

Reply:
To be solved in a new issue 👍

Reply:
I've opened an issue here: frontity/wp-org-connection#20